schema/deploy/function_set_random_id.sql

```diff
@@ -0,0 +1,44 @@
+-- Deploy ggircs-portal:function_set_random_id to pg
+-- requires: schema_ggircs_portal
+
+begin;
+create extension if not exists pgcrypto;
+create function ggircs_portal.set_random_id()
+  returns trigger as $$
+
+declare
+  new_id varchar(1000);
+begin
+  while new_id in (select id from ggircs_portal.certification_url) or new_id is null loop
+    if new_id in (select id from ggircs_portal.certification_url) then
```

Comment:
This is currently running the select statement twice per iteration of the loop.

Have a look at [`loop` and `exit`](https://www.postgresql.org/docs/current/plpgsql-control-structures.html#PLPGSQL-CONTROL-STRUCTURES-LOOPS) for this?